### PR TITLE
WIP: Use auto args and config

### PR DESCRIPTION
**Pull request by jstar1**

### CODE REVIEW

File: dffml/df/memory.py [hunk 1]
```diff
@@ -1065,7 +1072,8 @@ class MemoryOrchestratorConfig(BaseOrchestratorConfig):
     """
 
 
-class MemoryOrchestratorContextConfig(NamedTuple):
+@config
+class MemoryOrchestratorContextConfig:
```
Comment (pdxjohnny) [inline dffml/df/memory.py]: We'll need to add

```python
@entrypoint("memory")
class MemoryOrchestrator(BaseOrchestrator, BaseMemoryDataFlowObject):
    CONFIG = MemoryOrchestratorConfig
    CONTEXT = MemoryOrchestratorContext
```

You'll also need to set the `CONFIG` property for all the non-`Context` classes. So anything that is a subclass of `BaseMemoryDataFlowObject`.

File: dffml/df/memory.py [hunk 2]
```diff
@@ -205,7 +209,8 @@ def __call__(self) -> NotificationSetContext:
         return NotificationSetContext(self)
 
 
-class MemoryInputNetworkContextEntry(NamedTuple):
+@config
+class MemoryInputNetworkContextEntry:
```
Comment (pdxjohnny) [inline dffml/df/memory.py]: Keep this as `NamedTuple`, let's only changed classes that end in `Config`

File: dffml/df/memory.py (hunk 1 above)
Comment (jstar1) [reply dffml/df/memory.py]: Just to clarify is this required for this merge request?

Comment (pdxjohnny) [reply dffml/df/memory.py]: Ya, we'll need to remove the `args()` and `config()` methods as well, and this is what takes their place.

Comment (jstar1) [reply dffml/df/memory.py]: Got it thanks @pdxjohnny

**Pull request merged**